base_revision/models/base_revision.py

```diff
@@ -96,13 +96,14 @@ def copy_revision_with_context(self):
         self.write(self._prepare_revision_data(new_revision))
         return new_revision
 
-    @api.model
-    def create(self, values):
-        rec = super().create(values)
-        if "unrevisioned_name" not in values:
-            name_field = self._context.get("revision_name_field", "name")
-            rec.write({"unrevisioned_name": rec[name_field]})
-        return rec
+    @api.model_create_multi
+    def create(self, vals_list):
+        records = super().create(vals_list)
+        name_field = self._context.get("revision_name_field", "name")
+        for rec, values in zip(records, vals_list):
+            if "unrevisioned_name" not in values:
+                rec.write({"unrevisioned_name": rec[name_field]})
```

Comment:
As you are here, why not doing this before calling super for not having 2 DB operations per record.

Reply:
Can we be sure that the "name" value is set in the vals_list? Also in case of e.g. invoices or sales orders, where the name is generated with a continues number?

Reply:
You're welcome, and I also appreciate your will to improve existing code and to receive positively the comments.